### PR TITLE
Add missing gettext calls to site settings

### PR DESCRIPTION
**Pull request by th3hamm0r**
_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/contrib/settings/models.py [hunk 1]
```diff
@@ -144,7 +145,10 @@ def for_site(cls, site):
         return instance
 
     def __str__(self):
-        return "%s for %s" % (self._meta.verbose_name.capitalize(), self.site)
+        return _("%(site_setting)s for %(site)s") % {
+            "site_setting": self._meta.verbose_name.capitalize(),
```
Comment (lb-) [inline wagtail/contrib/settings/models.py:149]: as per discussion on https://github.com/wagtail/wagtail/issues/9219

let's pull out this `capitalize` call and either a. replace with capfirst OR find the location in the template and update with `|capfirst` filter (ideal if not too complex).

File: wagtail/contrib/settings/templates/wagtailsettings/edit.html [hunk 2]
```diff
@@ -15,7 +15,8 @@
             </form>
         {% endif %}
     {% endfragment %}
-    {% include "wagtailadmin/shared/header.html" with title="Editing" icon="cogs" subtitle=setting_type_name|capfirst merged=1 extra_actions=editing_actions %}
+    {% trans "Editing" as editing_str %}
+    {% include "wagtailadmin/shared/header.html" with title=editing_str icon="cogs" subtitle=setting_type_name|capfirst merged=1 extra_actions=editing_actions %}
```
Comment (lb-) [inline wagtail/contrib/settings/templates/wagtailsettings/edit.html:19]: note - I can see here we are already using `|capfirst` but for a similar purpose.

File: wagtail/contrib/settings/templates/wagtailsettings/edit.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {% extends "wagtailadmin/base.html" %}
 {% load i18n wagtailadmin_tags %}
 
-{% block titletag %}{% blocktrans trimmed %}Editing {{ instance }}{% endblocktrans %}{% endblock %}
+{% block titletag %}{% blocktrans trimmed with instance=instance|capfirst %}Editing {{ instance }}{% endblocktrans %}{% endblock %}
```
Comment (th3hamm0r) [inline wagtail/contrib/settings/templates/wagtailsettings/edit.html:4]: Fyi: I was not 100% sure if I should change the format of the string, but I've kept it, so it won't create a new string id to translate.

Comment (lb-) [reply wagtail/contrib/settings/templates/wagtailsettings/edit.html:4]: good call

**Review by lb- — APPROVED** [review]
Found another un-translated label - will fix on your PR
https://github.com/wagtail/wagtail/blob/6a8ce3161ec4b985b5f199aaf4f67a23c647589e/wagtail/contrib/settings/templates/wagtailsettings/index.html#L7

**Pull request merged**